services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy

```diff
@@ -64,25 +64,79 @@ class SwitchRulesSpec extends BaseSpecification {
         }
     }
 
-    @Ignore("Test is skipped because of the issue #1464")
-    def "Pre-installed rules are not deleted from a new switch when connecting it to the controller"() {
-        // TODO(ylobankov): Also the current disconnection of the switch from the controller is not suitable for
-        // this scenario. It is needed to add additional functionality to properly disconnect the switch.
+    def "Pre-installed rules (including default) are not deleted from a new switch connected to the controller"() {
+        requireProfiles("virtual")
+
+        given: "A switch with some rules installed (including default) and not connected to the controller"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.getActiveSwitches()[0..1]
+        def defaultRules = northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.sort { it.cookie }
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        northboundService.addFlow(flow)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        def defaultPlusFlowRules = []
+        assert Wrappers.wait(WAIT_OFFSET) {
+            defaultPlusFlowRules = northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.sort { it.cookie }
+            defaultPlusFlowRules.size() == defaultRules.size() + FLOW_RULES_SIZE
+        }
+
+        lockKeeperService.knockoutSwitch(srcSwitch.dpId)
+        assert Wrappers.wait(discoveryTimeout + rerouteDelay + WAIT_OFFSET * 2) {
+            northboundService.getFlowStatus(flow.id).status == FlowState.DOWN
```

Comment:
Do you really have to wait until it goes down? You can try removing the flow as soon as switch goes down (do not wait for flow reroute), this should be possible

Reply:
Ok, will fix.